### DISCUSSION TITLE
recent log4j critical vulnerability

### DISCUSSION THREAD

**Discussion by ivkina** [eclipse-birt/birt#776] [Q&A] [answered]
Is BIRT runtime/designer/eclipse affected? They say the issue is critical and requires immediate patching/fixing.

https://logging.apache.org/log4j/2.x/security.html
Fixed in Log4j 2.15.0
CVE-2021-44228: Apache Log4j2 JNDI features do not protect against attacker controlled LDAP and other JNDI related endpoints.

Severity: Critical

Base CVSS Score: 10.0 CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H

Versions Affected: all log4j-core versions >=2.0-beta9 and <=2.14.1

Descripton: Apache Log4j <=2.14.1 JNDI features used in configuration, log messages, and parameters do not protect against attacker controlled LDAP and other JNDI related endpoints. An attacker who can control log messages or log message parameters can execute arbitrary code loaded from LDAP servers when message lookup substitution is enabled. From log4j 2.15.0, this behavior has been disabled by default.

Mitigation: In releases >=2.10, this behavior can be mitigated by setting either the system property log4j2.formatMsgNoLookups or the environment variable LOG4J_FORMAT_MSG_NO_LOOKUPS to true. For releases >=2.7 and <=2.14.1, all PatternLayout patterns can be modified to specify the message converter as %m{nolookups} instead of just %m. For releases >=2.0-beta9 and <=2.10.0, the mitigation is to remove the JndiLookup class from the classpath: zip -q -d log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class.

**Comment by SteveSchafer-Innovent** [comment]
There's no log4j jar in the runtime but that's not a guarantee the classes are not loaded by some other means.  Can anyone eliminate that as a possibility?

**Reply by wimjongman** [reply]
There could be jars-in-jars (fat jars). We need to know two things:

1. Are we using log4j? The best is to scan our java sources on the usage of log4j. 
2. Are our dependencies using log4j? For this we need to unpack all jars/zips and find look for log4j

**Comment by wimjongman** [comment, marked as answer]
I scanned the sources and we do not use org.apache.logging.log4j anywhere.

https://wiki.eclipse.org/Eclipse_and_log4j2_vulnerability_(CVE-2021-44228)

**Reply by SteveSchafer-Innovent** [reply]
I have also heard that OpenText has said they do not use it in their commercial offering.

**Reply by hvbtup** [reply]
I wrote a tool to look recursively into all those ZIP and JAR files we deliver to our clients (including a slightly modified BIRT POJO runtime) and didn't find any usage of log4j2.

**Reply by wimjongman** [reply]
Thanks, Henning. Maybe you can make a website entry regarding this CVE?

**Reply by hvbtup** [reply]
@wimjongman What for? My comment only means that not only you scanned the latest sources, but also I scanned the generated BIRT POJO runtime (of an older release, however) including all those libraries. 
This doesn't say anything about the generated all-in-one designer, for example.
So I don't have any additional information - it is just a second pair of eyes that confirms your statement that BIRT doesn't use log4j.

IMHO it should suffice to add the CVE number to the title of this discussion.

BTW regarding the website: It is still difficult to find the new web-site. A direct hit finds the new website on github, but e.g. https://www.eclipse.org/birt/about/ still is old documentation - when you google for eclipse birt, you need some luck to find the new website.

**Reply by wimjongman** [reply]
> CVE

Ok, I agree.

> website

Yes, it seems that deep linking the website still resolves to the old website. I have filed this issue:

https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/779
